modules/ip/datapath/ip_input.c

```diff
@@ -181,3 +203,134 @@ GR_DROP_REGISTER(ip_input_bad_checksum);
 GR_DROP_REGISTER(ip_input_bad_length);
 GR_DROP_REGISTER(ip_input_bad_version);
 GR_DROP_REGISTER(ip_input_other_host);
+
+#ifdef __GROUT_UNIT_TEST__
+struct node_infos node_infos = STAILQ_HEAD_INITIALIZER(node_infos);
+mock_func(const struct nexthop *, fib4_lookup(uint16_t, ip4_addr_t));
+mock_func(void *, gr_mbuf_trace_add(struct rte_mbuf *, struct rte_node *, size_t));
+mock_func(uint16_t, drop_packets(struct rte_graph *, struct rte_node *, void **, uint16_t));
+mock_func(int, drop_format(char *, size_t, const void *, size_t));
+mock_func(int, trace_ip_format(char *, size_t, const struct rte_ipv4_hdr *, size_t));
+mock_func(void, gr_eth_input_add_type(rte_be16_t, const char *));
+mock_func(void, loopback_input_add_type(rte_be16_t, const char *));
+
+static void __wrap_rte_node_enqueue_x1(
+	struct rte_graph *graph,
+	struct rte_node *node,
+	rte_edge_t next,
+	void *obj
+) {
+	(void)graph;
+	(void)node;
```

Comment:
I'm not sure if having named arguments is required by cmocka. If not, use use unnamed args:

https://github.com/DPDK/grout/blob/main/CONTRIBUTING.md#unused-function-parameters